### PR TITLE
Improving the flow of the portfolio workshop

### LINKED ISSUE #247: workshop: Portfolio
Moving signing up to the top to avoid loosing work and confusion with Github and JSBin. @jonleung

### PR DESCRIPTION
**Pull request by jonleung**
Closes #247 and #267

### CODE REVIEW

**Comment by jonleung** [comment]
@zachlatta, do you have time to review this?

File: workshops/portfolio/README.md [hunk 1]
```diff
@@ -11,21 +11,70 @@ Your final design looks like this:
 
 > ![](img/final_1.png)
 
-Here is a link to a [live demo](http://output.jsbin.com/fugoki/2)
+Here is a link to a
+<a href="http://output.jsbin.com/fugoki/2" target="_blank">live demo</a>
```
Comment (zachlatta) [inline workshops/portfolio/README.md]: I don't think we should use `a` tags when we really don't have to, for the sake of Markdown. I'd revert this change.

File: workshops/portfolio/README.md [hunk 2]
```diff
@@ -11,21 +11,70 @@ Your final design looks like this:
...
+And here is the
+<a href="http://jsbin.com/fugoki/2/edit" target="_blank">final code</a>
+for the live demo.
 
 To do this, you will be learning the basics of two languages: HTML and CSS.
 
 Every website that you have ever seen are written in these two languages.
 
-## Microsoft Word for Code
+We will be building this website on on an online code editor called JS Bin.
```
Comment (zachlatta) [inline workshops/portfolio/README.md]: "on on"

File: workshops/portfolio/README.md [hunk 3]
```diff
@@ -478,60 +527,11 @@ Let's
...
-
-Now that you have a GitHub account, you can login with GitHub
-
-> ![](img/login_with_github.gif)
+Huzzah! We did it! We're done!
 
 ## Get a link that everyone can see
 
-Now that you're logged in, JS Bin will not "turn off" your website.
+Since you're logged in, JS Bin will not "turn off" your website.
```
Comment (zachlatta) [inline workshops/portfolio/README.md]: I personally prefer the "now that" wording. I think it's a bit clearer and less assuming (if that makes sense).

**Comment by zachlatta** [comment]
Sure!

File: workshops/portfolio/README.md [hunk 4]
```diff
@@ -11,21 +11,70 @@ Your final design looks like this:
...
 > ![](img/final_1.png)
 
-Here is a link to a [live demo](http://output.jsbin.com/fugoki/2)
+Here is a link to a
+<a href="http://output.jsbin.com/fugoki/2" target="_blank">live demo</a>
 
-And here is the [final code](http://jsbin.com/fugoki/2/edit) for the live demo.
+And here is the
+<a href="http://jsbin.com/fugoki/2/edit" target="_blank">final code</a>
+for the live demo.
```
Comment (zachlatta) [inline workshops/portfolio/README.md]: Same thing here.[

File: workshops/portfolio/README.md [hunk 5]
```diff
@@ -11,21 +11,70 @@ Your final design looks like this:
...
 
 To do this, you will be learning the basics of two languages: HTML and CSS.
 
 Every website that you have ever seen are written in these two languages.
 
-## Microsoft Word for Code
+We will be building this website on on an online code editor called JS Bin.
+
+## Creating a GitHub account
```
Comment (zachlatta) [inline workshops/portfolio/README.md]: Perhaps change this to "Create a GitHub account"? Less gerunds.

File: workshops/portfolio/README.md [hunk 6]
```diff
@@ -11,21 +11,70 @@ Your final design looks like this:
...
+
+Before we go to JS Bin, we want to have GitHub account.
```
Comment (zachlatta) [inline workshops/portfolio/README.md]: "to have GitHub account"?

File: workshops/portfolio/README.md [hunk 7]
```diff
@@ -11,21 +11,70 @@ Your final design looks like this:
...
+
+GitHub is a website used by many professional coders to collaborate on code.
+Think Dropbox, but for code.
+
+GitHub also has a login feature similar to how Facebook has "Facebook login".
+Let's make a GitHub account so we can login to JS Bin and many websites in the
+programming ecosystem!
+
+### 1) In a new window, open <a href="https://github.com" target="_blank">github.com</a>
```
Comment (zachlatta) [inline workshops/portfolio/README.md]: Same thing here about links.

File: workshops/portfolio/README.md [hunk 8]
```diff
@@ -11,21 +11,70 @@ Your final design looks like this:
...
+
+> ![](img/finish_signup.png)
+
+And now we're done creating a GitHub account!
+
+## Creating a JS Bin Account
+
+Now that we have a GitHub account:
+
+1. open to <a href="http://jsbin.com" target="_blank">jsbin.com</a>
```
Comment (zachlatta) [inline workshops/portfolio/README.md]: Same thing here with links.

File: workshops/portfolio/README.md [hunk 9]
```diff
@@ -11,21 +11,70 @@ Your final design looks like this:
...
+Now click "Authorize application"
+
+> ![](img/authorize_application.png)
+
+And we're all set!
+
+## JS Bin is like Microsoft Word for Code
 
 Just like you use Microsoft Word to write English, you can use JS Bin to write
 code.
```
Comment (zachlatta) [inline workshops/portfolio/README.md]: The previous text uses "we" instead of "you". Which are we using?

**Comment by zachlatta** [comment]
Left a bunch of comments.

**Comment by jonleung** [comment]
Thanks @zachlatta

**Comment by hellyeah** [comment]
Should we update this branch and merge it?

File: workshops/portfolio/README.md (hunk 1 above)
Comment (jonleung) [inline workshops/portfolio/README.md]: Multiple club leaders report that members will click on the links that take them away from the readme and not be able to get back to the readme. Having this open in a new tab solves this problem.

I am opting to keep this as it is.

File: workshops/portfolio/README.md (hunk 2 above)
Comment (jonleung) [inline workshops/portfolio/README.md]: :+1:

File: workshops/portfolio/README.md (hunk 3 above)
Comment (jonleung) [inline workshops/portfolio/README.md]: Huh, I actually really like that!

:+1:

File: workshops/portfolio/README.md (hunk 4 above)
Comment (jonleung) [inline workshops/portfolio/README.md]: Keeping the target="_blank" for the reasons described above.

File: workshops/portfolio/README.md (hunk 5 above)
Comment (jonleung) [inline workshops/portfolio/README.md]: I prefer "Creating" because "Create" sounds like an action that needs to be taken when you read it as opposed to a title of a section.

File: workshops/portfolio/README.md (hunk 6 above)
Comment (jonleung) [inline workshops/portfolio/README.md]: Changed it to:

> Before we go to JS Bin, we want a GitHub account that we can use to signin with.

File: workshops/portfolio/README.md (hunk 7 above)
Comment (jonleung) [inline workshops/portfolio/README.md]: Same response as before.

File: workshops/portfolio/README.md (hunk 8 above)
Comment (jonleung) [inline workshops/portfolio/README.md]: Same response as before.

File: workshops/portfolio/README.md (hunk 9 above)
Comment (jonleung) [inline workshops/portfolio/README.md]: Changed to `we`.

**Comment by jonleung** [comment]
@zachlatta made all the changes except for getting rid of the `a` tags. I explain why in a comment. Assigning back to you : )

**Comment by zachlatta** [comment]
Cool, can you update this branch and I'll merge?

**Comment by jonleung** [comment]
@MaxWofford I'm not quite sure why this test failed?

**Comment by MaxWofford** [comment]
@jonleung That's my bad. They should pass in https://github.com/hackedu/markcop/pull/29
